### ISSUE TITLE
[BUG]: Thread leak in statistics

### ISSUE THREAD

**Issue by NikSimDev** [lev2p1/brazier#134]
### Version

v0.1.5

### Severity

Critical (crash, data loss)

### Problem Description

```cpp
// Server.cpp
std::thread stats_thread([this] {
    while (true) {  // Infinite loop!
        std::this_thread::sleep_for(10s);
        Logger::log(...);
    }
});
stats_thread.detach();  // The thread lives forever
```
**Problem:** When `stop()` is called, this thread will not terminate, causing undefined behavior.

**Solution:**
```cpp
std::atomic<bool> shutdown_flag{false};
auto stats_thread = std::thread([this] {
    while (!shutdown_flag) {
        std::this_thread::sleep_for(10s);
        Logger::log(...);
    }
});

void Server::stop() {
    shutdown_flag = true;
    work_guard_.reset();
    io_.stop();
    if (stats_thread.joinable()) stats_thread.join();
}
```

### Verification

- [x] I have searched for existing issues
- [x] I am using the latest version
- [x] I have provided a minimal reproduction
